src/detectors/HCAL/ProtoCluster_factory_HcalBarrelIslandProtoClusters.h

```diff
@@ -38,9 +38,11 @@ class ProtoCluster_factory_HcalBarrelIslandProtoClusters : public JFactoryT<edm4
         //   32 - number of sectors
         //   2  - number of rows per sector
         //   24 - number of towers per row
+        //   5  - number of tiles per tower
         u_adjacencyMatrix =
           "("
-          "  abs(fmod(tower_1, 24) - fmod(tower_2, 24))"
+          "  abs(fmod(tile_1, 5) - fmod(tile_2, 5))"
+          "  + abs(fmod(tower_1, 24) - fmod(tower_2, 24))"
           "  + min("
           "      abs((sector_1 - sector_2) * 2 + floor(tower_1 / 24) - floor(tower_2 / 24)),"
           "      32 * 2 - abs((sector_1 - sector_2) * 2 + floor(tower_1 / 24) - floor(tower_2 / 24))"
```

Comment:
```suggestion
                    "      abs(((sector_1 - sector_2) * 2 + floor(tower_1 / 24) - floor(tower_2 / 24)) * 5 + fmod(tile_1, 5) - fmod(tile_2, 5)),"
          "      32 * 2 - abs(((sector_1 - sector_2) * 2 + floor(tower_1 / 24) - floor(tower_2 / 24)) * 5 + fmod(tile_1, 5) - fmod(tile_2, 5))"
```

Reply:
I checked this for a few cases, and a think the `sector_1 - sector_2` terms need an extra 5 in front... For example, as written the expression works out to 7 for `(tile, tower, sector) = (4, 27, 20)` vs. `(0, 3, 1)` (which should be adjacent), and it works out to 71 for `(4, 47, 31)` vs. `(0, 23, 0)` (which also should be adjacent).

Adjusting it to `2 * 5 * abs(sector_1 - sector_2)`  makes both cases to 1!

Reply:
Yes, there were brackets: `(x * 2 + y) * 5 + z`